recipes/microprofile/all/conanfile.py

```diff
@@ -0,0 +1,222 @@
+    topics = ("conan", "profiler", "embedded", "timer")
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "enabled": [True, False],
+        "debug": [True, False],
+        "use_miniz": [True, False],
+        "thread_buffer_size": "ANY",
+        "thread_gpu_buffer_size": "ANY",
+        "max_frame_history": "ANY",
+        "webserver_port": "ANY",
+        "webserver_maxframes": "ANY",
+        "webserver_socket_buffer_size": "ANY",
+        "gpu_frame_delay": "ANY",
+        "name_max_length": "ANY",
+        "max_timers": "ANY",
+        "max_threads": "ANY",
+        "max_string_length": "ANY",
+        "timeline_max_tokens": "ANY",
+        "thread_log_frames_reuse": "ANY",
+        "max_groups": "ANY",
+        "use_big_endian": [True, False],
+        "enable_gpu_timer_callbacks": [True, False],
+        "enable_gl_timers": [True, False],
+        "enable_d3d11_timers": [True, False],
+        "enable_d3d12_timers": [True, False],
+        "enable_vulkan_timers": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "enabled": True,
+        "debug": False,
+        "use_miniz": False,
+        "thread_buffer_size": 2048 << 10,
+        "thread_gpu_buffer_size": 128 << 10,
+        "max_frame_history": 512,
+        "webserver_port": 1338,
+        "webserver_maxframes": 30,
+        "webserver_socket_buffer_size": 16 << 10,
+        "gpu_frame_delay": 5,
+        "name_max_length": 64,
+        "max_timers": 1024,
+        "max_threads": 32,
+        "max_string_length": 128,
+        "timeline_max_tokens": 64,
+        "thread_log_frames_reuse": 200,
+        "max_groups": 128,
+        "use_big_endian": False,
+        "enable_gpu_timer_callbacks": False,
+        "enable_gl_timers": False,
+        "enable_d3d11_timers": False,
+        "enable_d3d12_timers": False,
+        "enable_vulkan_timers": False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        else:
+            del self.options.enable_d3d11_timers
+            del self.options.enable_d3d12_timers
+
+    def _validate_max_groups(self):
+        try:
+            _max_groups = int(self.options.max_groups)
+            if _max_groups % 32 != 0:
+                raise ConanInvalidConfiguration("max_groups must be multiple of 32.")
+        except ValueError:
+            raise ConanInvalidConfiguration("max_groups must be a number.")
+
+    def _validate_timers(self):
+        _timers_enabled_count = 0
```

Comment:
Do we care if some option that expects an integer gets passed a random string or negative integer?

Reply:
Not sure what you are referring to, but I have changed this timers option to a list and simplified the validation function.

Reply:
You have a lot of options with `ANY`.
Nothing prevents me to pass a non-integer string.

Reply:
Oh I get it now. Yes they should be checked as the library would most likely not even compile with invalid values. If I validate each `ANY` option in the `validate()` function, will it be okay? Or is there a better way to validate them?

Reply:
You can use a for loop to minimize code duplication and possible errors.
e.g.
```python
positive_int_options = [
    "option1",
    "option2",
    "option3",
    ...
]
for opt in positive_int_options:
    try:
        value = int(getattr(self.options, opt))
        if value < 0:
            raise ValueError
    except ValueError:
        raise ConanInvalidConfiguration("microprofile:{} must be a positive integer".format(opt))
```